### PR TITLE
Add changes to add the new setting in SCA

### LINKED ISSUE #5652: Add changes to SCA documentation to add new regex implementation
## Description
This issue aims to perform an editorial and grammar review of the following sections of `Security Configuration Assessment`:
- Modify the documentation to accommodate the new regex engine information.

## Tasks:
- [x] Work on branch `5652-add-changes-to-SCA-documentation`
- [x] perform an editorial and grammar review of:
   - [x] Changes on Security Configuration Assessment 
- [x] Work on PR
- [x] Ask for a review.

### PR DESCRIPTION
**Pull request by FrancoRivero**
## Description
Hi team, this pull-request is to modify the SCA documentation as we have added a new setting that makes it possible to add rules using the `PCRE2` engine to detect and resolve regex checks.
Regards

## Checks
- [x] It compiles without warnings.
- [x] Spelling and grammar. 
- [x] Used impersonal speech. 
- [x] Used uppercase only on nouns.

Closes #5652

### CODE REVIEW

File: source/user-manual/capabilities/sec-config-assessment/creating-custom-policies.rst [hunk 1]
```diff
@@ -103,6 +103,15 @@ Each section has its fields as described in the tables
...
     | references         | No             | Array of strings  | Any string             | Any string             |
     +--------------------+----------------+-------------------+------------------------+------------------------+
+    | regex_type         | No             | String            | "pcre2" or "osregex"   | Policy regex engine    |
+    +--------------------+----------------+-------------------+------------------------+------------------------+
+
+
+.. note::
+  If the **regex_type** is configured in this section the checks will be run with the configured engine.
+
+.. note::
```
Comment (chemamartinez) [inline source/user-manual/capabilities/sec-config-assessment/creating-custom-policies.rst:113]: We can merge both notes and explain this easier.

**Review by chemamartinez — APPROVED** [review]
LGTM

File: source/user-manual/capabilities/sec-config-assessment/creating-custom-policies.rst [hunk 2]
```diff
@@ -181,10 +183,16 @@ Each check is comprised by several fields as described in the table :ref:`sca_po
...
     |    rules    |    Yes    |      Array of strings      |     Any string     |
     +-------------+-----------+----------------------------+--------------------+
+    | regex_type  |    No     |           String           |"pcre2" or "osregex"|
+    +-------------+-----------+----------------------------+--------------------+
+
 
 Check evaluation is governed by its `rule result aggregation strategy`, as set in its ``condition`` field, and the results of
 the evaluation of its rules.
 
+.. note::
```
Comment (javimed) [inline source/user-manual/capabilities/sec-config-assessment/creating-custom-policies.rst:193]:
```suggestion
.. note::

```

File: source/user-manual/capabilities/sec-config-assessment/creating-custom-policies.rst [hunk 3]
```diff
@@ -181,10 +183,16 @@ Each check is comprised by several fields as described in the table :ref:`sca_po
     +-------------+-----------+----------------------------+--------------------+
     |    rules    |    Yes    |      Array of strings      |     Any string     |
     +-------------+-----------+----------------------------+--------------------+
+    | regex_type  |    No     |           String           |"pcre2" or "osregex"|
+    +-------------+-----------+----------------------------+--------------------+
+
```
Comment (javimed) [inline source/user-manual/capabilities/sec-config-assessment/creating-custom-policies.rst:188]:
```suggestion
```

File: source/user-manual/capabilities/sec-config-assessment/creating-custom-policies.rst [hunk 4]
```diff
@@ -181,10 +183,16 @@ Each check is comprised by several fields as described in the table :ref:`sca_po
...
 
 Check evaluation is governed by its `rule result aggregation strategy`, as set in its ``condition`` field, and the results of
 the evaluation of its rules.
 
+.. note::
+  If the **regex_type** is configured for a particular check, it overwrites the regex engine applied for the whole policy.
```
Comment (javimed) [inline source/user-manual/capabilities/sec-config-assessment/creating-custom-policies.rst:194]:
```suggestion
   If you set a *regex_type*, it overrides the regex engine type defined in the policy.
```

**Review by javimed — CHANGES_REQUESTED** [review]

**Review by javimed — APPROVED** [review]
LGTM

**Pull request merged**